app/src/main/java/com/example/cinemapp/data/MovieRepository.kt

```diff
@@ -30,6 +39,51 @@ class MovieRepository(
         }
     }
 
+    suspend fun getCredits(movieId: Int): MovieCreditsDTO? {
```

Comment:
Try reusing the logic for fetching and returning data from remote data source, since you have the same logic on 3 places in this repository.

For example, you can do this:

`
fun <T> responseBodyFrom(response: Response<T>): T? {
    return try {
        if (response.isSuccessful) {
            response.body()
        } else {
            Log.e(TAG, response.message())
            null
        }
    } catch (e: Exception) {
        Log.e(TAG, e.message ?: "Unknown error")
        null
    }
}
`

and then have something like:

`suspend fun getCredits(movieId: Int): MovieCreditsDTO? {
    return responseBodyFrom(remoteDataSource.getMovieCredits(movieId = movieId))
}
`